src/Sylius/Behat/Context/Api/Admin/ManagingTaxonsContext.php

```diff
@@ -89,4 +93,74 @@ public function iShouldSeeTheTaxonNamedAnd(...$expectedTaxonNames): void
 
         Assert::allOneOf($expectedTaxonNames, $taxonNames);
     }
+
+    /**
+     * @When I remove taxon named :name
+     * @When I delete taxon named :name
+     * @When I try to delete taxon named :name
+     */
+    public function iRemoveTaxonNamed(string $name): void
+    {
+        $code = StringInflector::nameToLowercaseCode($name);
+
+        $this->apiClient->delete(Resources::TAXONS, $code);
+    }
+
+    /**
+     * @Then /^taxon named "([^"]+)" should not be added$/
+     * @Then the taxon named :name should no longer exist in the registry
+     */
+    public function taxonNamedShouldNotBeAdded(string $name): void
+    {
+        $code = StringInflector::nameToLowercaseCode($name);
+
+        Assert::false(
+            $this->responseChecker->hasItemWithValue($this->apiClient->index(Resources::TAXONS), 'code', $code)
+        );
+    }
+
+    /**
+     * @Then /^the ("[^"]+" taxon) should appear in the registry$/
+     */
+    public function theTaxonShouldAppearInTheRegistry(TaxonInterface $taxon): void
+    {
+        Assert::true(
+          $this->responseChecker->hasItemWithValue($this->apiClient->index(Resources::TAXONS), 'code', $taxon->getCode())
+        );
+    }
+
+    /**
+     * @Then I should be notified that I cannot delete a menu taxon of any channel
+     */
+    public function iShouldBeNotifiedThatICannotDeleteAMenuTaxonOfAnyChannel(): void
+    {
+        $statusCode = $this->apiClient->getLastResponse()->getStatusCode();
+
+        Assert::same($statusCode, 422);
```

Comment:
`ResponseChecker::isDeletionSuccessfull` method should be used here

Reply:
fixed